src/aiodynamo/utils.py

```diff
@@ -132,3 +54,80 @@ def deserialize(value: Dict[str, Any], numeric_type: Callable[[str], Any]) -> An
     if tag == "M":
         return {k: deserialize(v, numeric_type) for k, v in val.items()}
     raise TypeError(f"Dynamodb type {tag} is not supported")
+
+
+NUMERIC_TYPES = int, float, decimal.Decimal
+
+
+def serialize(value: Any) -> Optional[Dict[str, Any]]:
+    """
+    Serialize a Python value to a Dynamo Value, removing empty strings.
+    """
+    tag_and_value = low_level_serialize(value)
+    if tag_and_value is not None:
+        return {tag_and_value[0]: tag_and_value[1]}
+    return None
+
+
+def low_level_serialize(value: Any) -> Optional[Tuple[str, Any]]:
+    if value is None:
+        return "NULL", True
+    elif isinstance(value, bool):
+        return "BOOL", value
+    elif isinstance(value, NUMERIC_TYPES):
+        return "N", str(value)
+    elif isinstance(value, str):
+        if not value:
+            return None
+        return "S", value
+    elif isinstance(value, bytes):
+        if not value:
+            return None
+        return "B", base64.b64encode(value).decode("ascii")
+    elif isinstance(value, collections_abc.Set):
+        numeric_items, str_items, byte_items, total = reduce(
+            lambda acc, item: (
+                acc[0] + isinstance(item, NUMERIC_TYPES),
+                acc[1] + isinstance(item, str),
+                acc[2] + isinstance(item, bytes),
+                acc[3] + 1,
+            ),
+            value,
+            (0, 0, 0, 0),
+        )
+        if numeric_items == total:
+            return "NS", [str(item) for item in value]
+        elif str_items == total:
+            return "SS", [item for item in value if item]
+        elif byte_items == total:
+            return (
+                "BS",
+                [base64.b64encode(item).decode("ascii") for item in value if item],
+            )
+        else:
+            raise TypeError(
+                f"Sets which are not entirely numeric, strings or bytes are not supported. value: {value!r}"
+            )
+    elif isinstance(value, collections_abc.Mapping):
+        return "M", serialize_dict(value)
+    elif isinstance(value, list):
```

Comment:
I wonder if some `collections_abc.Xxx` would be better than literal `list`, to cover, e.g. tuples and namedtuples and what not.

Reply:
🤔 `Sequence` might indeed be better. (`Iterable`/`Iterator` might be a bit too wild). I used `list` here because that's what boto3 uses...